reana_db/models.py

```diff
@@ -435,6 +476,42 @@ def can_transition_to(self, next_status):
         return current_transition in ALLOWED_WORKFLOW_STATUS_TRANSITIONS
 
 
+@event.listens_for(Workflow.status, "set")
+def workflow_status_change_listener(workflow, new_status, old_status, initiator):
+    """Workflow status change listener."""
+    if new_status in [
+        WorkflowStatus.finished,
+        WorkflowStatus.failed,
+    ]:
+        workflow.run_finished_at = datetime.now()
+    elif new_status in [WorkflowStatus.stopped]:
+        workflow.run_stopped_at = datetime.now()
+
+    finished_at = workflow.run_finished_at or workflow.run_stopped_at
+    if workflow.run_started_at and finished_at:
+        cpu_time = finished_at - workflow.run_started_at
+        cpu_milliseconds = int(cpu_time.total_seconds() * 1000)
+        cpu_resource = Resource.query.filter_by(
+            name=DEFAULT_QUOTA_RESOURCES["cpu"]
+        ).one_or_none()
```

Comment:
Just asking for confirmation. The disk quota is going to be calculated nightly (creation of `WorkflowResource`s and update of `UserResource`)? What if the quota is exceeded before calculated?

Reply:
I think that the [daemon](https://github.com/reanahub/reana-workflow-controller/issues/333) should not be aware of the quota limitations because:
- It should be intelligent enough to not calculate the disk usage for something that hasn't changed (sort of caching)
- We will stop users from increasing their quota usage (https://github.com/reanahub/reana-server/issues/282 and https://github.com/reanahub/reana-server/issues/283), so the action that they will most probably take will be deletion of files and workflows [to recover quota](https://github.com/reanahub/reana-ui/issues/129), and in that case we really want to recalculate it and show it.

The problem I see here though is that disk quota won't be recalculated on-demand/live. That has two solutions:
1. Recalculate on every action of type [deletion, overwriting ...], probably very expensive
2. Add a new functionality for users to trigger the recalculation of their disk usage

What do you think? cc @tiborsimko @audrium